src/test/java/io/jenkins/plugins/coverage/model/CoverageEvaluatorTest.java

```diff
@@ -0,0 +1,174 @@
+        evaluator.add(new QualityGate(CoverageMetric.METHOD, 85, QualityGateStatus.FAILED));
+        evaluator.add(new QualityGate(CoverageMetric.METHOD, 90, QualityGateStatus.WARNING));
+        QualityGateStatus buildStatus = evaluator.evaluate(node, logger);
+        assertThat(buildStatus).isEqualTo(QualityGateStatus.FAILED);
+        String[] loggerValues
+                   = {"-> NOT PASSED: FAILED - Quality Gate: (FAILED, 85.000, Method); ACHIEVED: 80.000"};
+        assertThat(logger.getMessages()).containsExactlyElementsOf(Arrays.asList(loggerValues));
+        logger.clear();
+    }
+
+    @Test
+    void shouldReturnInactive() {
+        Logger logger = new Logger();
+        CoverageNode node = getCoverageNode();
+        CoverageEvaluator evaluator = new CoverageEvaluator();
+        QualityGateStatus buildStatus = evaluator.evaluate(node, logger);
+        assertThat(buildStatus).isEqualTo(QualityGateStatus.INACTIVE);
+        String[] loggerValues
+                   = {"-> INACTIVE - No quality gate defined"};
+
+        assertThat(logger.getMessages()).containsExactlyElementsOf(Arrays.asList(loggerValues));
+        logger.clear();
+    }
+
+    @Test
+    void shouldAddOneQualityGateToEmptyList() throws QualityGatesInvalidException {
+        CoverageEvaluator evaluator = new CoverageEvaluator();
+        QualityGate gate = new QualityGate(CoverageMetric.LINE, 50, QualityGateStatus.WARNING);
+        evaluator.add(gate);
+        assertThat(evaluator.getQualityGateSet()).hasSize(1);
+    }
+
+    @Test
+    void shouldAddThreeOutOfFourQualityGatesFromList() throws QualityGatesInvalidException {
+        CoverageEvaluator evaluator = new CoverageEvaluator();
+        evaluator.addAll(Arrays.asList(new QualityGate(CoverageMetric.LINE, 50, QualityGateStatus.WARNING),
+                new QualityGate(CoverageMetric.LINE, 30, QualityGateStatus.WARNING),
+                new QualityGate(CoverageMetric.BRANCH, 10, QualityGateStatus.WARNING),
+                new QualityGate(CoverageMetric.BRANCH, 10, QualityGateStatus.FAILED)));
+        assertThat(evaluator.getQualityGateSet()).hasSize(3);
+    }
+
+    @Test
+    void shouldRemoveAQualityGate() throws QualityGatesInvalidException {
+        CoverageEvaluator evaluator = new CoverageEvaluator();
+        QualityGate gate = new QualityGate(CoverageMetric.LINE, 50, QualityGateStatus.WARNING);
+        evaluator.add(gate);
+        evaluator.remove(gate);
+        assertThat(evaluator.getQualityGateSet()).isEmpty();
+    }
+
+    private static class Logger implements FormattedLogger {
+        private final List<String> messages = new ArrayList<>();
+
+        @Override
+        public void print(final String format, final Object... args) {
+            messages.add(String.format(format, args));
+        }
+
+        List<String> getMessages() {
+            return messages;
+        }
+
+        void clear() {
+            messages.clear();
+        }
+    }
+
+
+}
```

Comment:
In total, the tests are well written and cover most important paths. (One is missing: a quality gate that has no match in the node).